### ISSUE TITLE
Cholmod version is recorded somewhere in source?

### ISSUE THREAD

**Issue by fxcoudert** [JuliaSparse/SparseArrays.jl#359]
SuiteSparse update in Julia in https://github.com/JuliaLang/julia/pull/48855 is blocked because somehow an old cholmod version is recorded, apparently in this repo:

```
┌ Warning: CHOLMOD version incompatibility
  | │
  | │ Julia was compiled with CHOLMOD version 3.0.14. It is
  | │ currently linked with version 4.0.3.
  | │ This might cause Julia to terminate when working with
  | │ sparse matrix factorizations, e.g. solving systems of
  | │ equations with \.
  | │
  | │ It is recommended that you use Julia with the same major
  | │ version of CHOLMOD as the one used during the build, or
  | │ download the generic binaries from www.julialang.org,
  | │ which ship with the correct versions of all dependencies.
  | └ @ SparseArrays.CHOLMOD /cache/build/default-amdci5-7/julialang/juliamaster/usr/share/julia/stdlib/v1.10/SparseArrays/src/solvers/cholmod.jl:199
```

I do not understand what “Julia was compiled with CHOLMOD version 3.0.14” means. In the PR, it is absolutely wrong: Julia was compiled with version 4.0.3, the PR is updating that version.

The warning makes several test fail (output matching becomes wrong).

**Comment by ViralBShah** [comment]
We need to update the SuiteSparse auto-generated wrappers in `gen`. The script doesn't run right now, because there seems to have been a change in the `.h` filenames and how they are organized.
```
julia> include("generator.jl")
  Downloaded artifact: SuiteSparse
ERROR: LoadError: AssertionError: isfile(cholmod_blas_h)
```

**Comment by Gnimuc** [comment]
Is this expected? 

https://github.com/JuliaSparse/SparseArrays.jl/actions/runs/4492043256/jobs/7901368283?pr=363

**Comment by ViralBShah** [comment]
Right - that needs updating of the wrappers. It is picking up the old ones.

**Comment by ViralBShah** [comment]
Being solved in #363
